### PR TITLE
Update Kestrel Host Filtering Middleware guidance

### LINKED ISSUE #5812: Update Kestrel host configuration guidance to include new HostFiltering middleware
In 2.0 we issued a patch and guidance related to host configuration, especially for Kestrel that doesn't have built in support for host filtering.
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel?tabs=aspnetcore2x#host-filtering

In 2.1 preview2 we've taken that sample and created a new HostFiltering package & middleware. That's been added to WebHost.CreateDefaultBuilder and can be configured by setting "AllowedHosts" to a semicolon separated list like "locahost;example.com". The middleware is disabled by default.
RE: https://github.com/aspnet/MetaPackages/pull/261

Update the Kestrel docs accordingly.

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #5812 

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel?view=aspnetcore-1.0&branch=pr-en-us-6643)

Thanks @RehanSaeed for your Q&A chat, which improved these updates.

@Tratcher Do we need any kind of wildcard binding warning here?

### CODE REVIEW

File: aspnetcore/fundamentals/servers/kestrel.md [hunk 1]
```diff
@@ -798,23 +803,40 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
...
-```json
-{
-  "AllowedHosts": "example.com"
-}
-```
+::: moniker-end
+
+::: moniker range=">= aspnetcore-2.1"
+
+As a workaround, use Host Filtering Middleware. Host Filtering Middleware is provided by the [Microsoft.AspNetCore.HostFiltering](https://www.nuget.org/packages/Microsoft.AspNetCore.HostFiltering) package, which is included in the [Microsoft.AspNetCore.App](xref:fundamentals/metapackage-app) metapackage. The middleware is added by [CreateDefaultBuilder](/dotnet/api/microsoft.aspnetcore.webhost.createdefaultbuilder), which calls [AddHostFiltering](/dotnet/api/microsoft.aspnetcore.builder.hostfilteringservicesextensions.addhostfiltering):
```
Comment (Tratcher) [inline aspnetcore/fundamentals/servers/kestrel.md]: use _the_ Host Filtering Middleware

File: aspnetcore/fundamentals/servers/kestrel.md [hunk 2]
```diff
@@ -798,23 +803,40 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
...
 
 ```json
 {
-  "AllowedHosts": "localhost"
+  "AllowedHosts": "example.com;localhost"
 }
 ```
 
+> [!NOTE]
+> Don't use Host Filtering Middleware and [Forwarded Headers Middleware](xref:host-and-deploy/proxy-load-balancer) ([ForwardedHeadersOptions.AllowedHosts](/dotnet/api/microsoft.aspnetcore.builder.forwardedheadersoptions.allowedhosts)) simultaneously. Host Filtering Middleware is appropriate when Kestrel is used as an edge server. Forwarded Headers Middleware is appropriate for reverse proxy server and load balancer host configurations.
```
Comment (Tratcher) [inline aspnetcore/fundamentals/servers/kestrel.md]: "Don't use simultaneously" is over-cautionary. "Note these middleware have similar functionality for different scenarios" is a more appropriate intro.

Comment (Tratcher) [reply aspnetcore/fundamentals/servers/kestrel.md]: You don't actually say here that the Forwarded Headers middleware also has an AllowedHosts option. Without that it's not obvious why you're mentioning that middleware.

Comment (guardrex) [reply aspnetcore/fundamentals/servers/kestrel.md]: That's why I have it parenthetically present ...

> [Forwarded Headers Middleware](xref:host-and-deploy/proxy-load-balancer) ([ForwardedHeadersOptions.AllowedHosts](/dotnet/api/microsoft.aspnetcore.builder.forwardedheadersoptions.allowedhosts)) 

... but I'll add it as a phrase on the next commit and see if that helps.

File: aspnetcore/fundamentals/servers/kestrel.md [hunk 3]
```diff
@@ -798,23 +803,40 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
...
 
 ```json
 {
-  "AllowedHosts": "localhost"
+  "AllowedHosts": "example.com;localhost"
 }
 ```
 
+> [!NOTE]
+> [Forwarded Headers Middleware](xref:host-and-deploy/proxy-load-balancer) has an [ForwardedHeadersOptions.AllowedHosts](/dotnet/api/microsoft.aspnetcore.builder.forwardedheadersoptions.allowedhosts) option. Forwarded Headers Middleware and Host Filtering Middleware have similar functionality for different scenarios. Setting `AllowedHosts` with Forwarded Headers Middleware is appropriate for reverse proxy server and load balancer host configurations. Setting `AllowedHosts` with Host Filtering Middleware is appropriate when Kestrel is used as an edge server.
```
[the remark this answers is not in the document]
Comment (guardrex) [reply aspnetcore/fundamentals/servers/kestrel.md]: Check the commit ... took your suggestions + added small phrase changes.

**Pull request merged**